awx/main/dispatch/worker/callback.py

```diff
@@ -98,7 +98,8 @@ def flush(self, force=False):
                                 # work around - if the host disappears before
                                 # the event can be processed
                                 e.host_id = None
-                            e.save()
+                            if not e.skip_save():
```

Comment:
I don't think we need this check;  if `skip_save()` is `False`, doesn't that mean the event won't have been placed into `self.buff` in the first place (see below)?

Reply:
I think you're right, I misunderstood what was happening here as a bailout from the bulk save.

Reply:
The main use case here is:

1.  Many events in one bulk save.
2.  One of the events has a `host_id` reference.
3.  That Host is actually deleted in a parallel request.
4.  `IntegrityError` on the bulk save.
5.  Fall back to (slow, but accurate) `for e in events: e.save()`